script/DeployCSVerifier.s.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: 2024 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+// Usage: forge script ./script/DeployCSVerifier:DeployCSVerifier[Holesky|Mainnet]
+
+pragma solidity 0.8.24;
+
+import { Script } from "forge-std/Script.sol";
+import { console2 as console } from "forge-std/console2.sol";
+
+import { CSVerifier } from "../src/CSVerifier.sol";
+import { GIndex } from "../src/lib/GIndex.sol";
+import { Slot } from "../src/lib/Types.sol";
+
+struct Config {
+    address withdrawalVault;
+    address module;
+    GIndex gIFirstWithdrawalPrev;
+    GIndex gIFirstWithdrawalCurr;
+    GIndex gIFirstValidatorPrev;
+    GIndex gIFirstValidatorCurr;
+    GIndex gIHistoricalSummariesPrev;
+    GIndex gIHistoricalSummariesCurr;
+    Slot firstSupportedSlot;
+    Slot pivotSlot;
+    uint64 slotsPerEpoch;
+}
+
+// Check the constants below via `yarn run gindex`.
+
+GIndex constant HISTORICAL_SUMMARIES_DENEB = GIndex.wrap(
+    0x0000000000000000000000000000000000000000000000000000000000003b00
+);
+GIndex constant FIRST_WITHDRAWAL_DENEB = GIndex.wrap(
+    0x0000000000000000000000000000000000000000000000000000000000e1c004
+);
+GIndex constant FIRST_VALIDATOR_DENEB = GIndex.wrap(
+    0x0000000000000000000000000000000000000000000000000056000000000028
+);
+
+GIndex constant HISTORICAL_SUMMARIES_ELECTRA = GIndex.wrap(
+    0x0000000000000000000000000000000000000000000000000000000000005b00
+);
+GIndex constant FIRST_WITHDRAWAL_ELECTRA = GIndex.wrap(
+    0x000000000000000000000000000000000000000000000000000000000161c004
+);
+GIndex constant FIRST_VALIDATOR_ELECTRA = GIndex.wrap(
+    0x0000000000000000000000000000000000000000000000000096000000000028
+);
+
+abstract contract DeployCSVerifier is Script {
+    Config internal config;
+
+    function run() public {
+        CSVerifier verifier = new CSVerifier({
+            withdrawalAddress: config.withdrawalVault,
+            module: config.module,
+            slotsPerEpoch: config.slotsPerEpoch,
+            gIFirstWithdrawalPrev: config.gIFirstWithdrawalPrev,
+            gIFirstWithdrawalCurr: config.gIFirstWithdrawalCurr,
+            gIFirstValidatorPrev: config.gIFirstValidatorPrev,
+            gIFirstValidatorCurr: config.gIFirstValidatorCurr,
+            gIHistoricalSummariesPrev: config.gIHistoricalSummariesPrev,
+            gIHistoricalSummariesCurr: config.gIHistoricalSummariesCurr,
+            firstSupportedSlot: config.firstSupportedSlot,
+            pivotSlot: config.pivotSlot
+        });
+        console.log("CSVerifier deployed at:", address(verifier));
+    }
+}
+
+contract DeployCSVerifierHolesky is DeployCSVerifier {
+    constructor() {
+        config = Config({
+            withdrawalVault: 0xF0179dEC45a37423EAD4FaD5fCb136197872EAd9,
```

Comment:
can we load corresponding main deploy config file?
for calling locator and using deployed module address here

Reply:
I don't think it makes a lot of sense. It adds a lot of complexity for something that can be easily verified.